### PR TITLE
add support for o1/o1-mini ENG-1114

### PR DESCRIPTION
**Pull request by r0ymanesco**
Add support for o1 and o1-mini

### CODE REVIEW

File: notdiamond/llms/providers.py [hunk 1]
```diff
@@ -23,6 +23,10 @@ class NDLLMProviders(Enum):
         GPT_4o_MINI_2024_07_18 (NDLLMProvider): refers to 'gpt-4o-mini-2024-07-18' model by OpenAI
         GPT_4o_MINI (NDLLMProvider): refers to 'gpt-4o-mini' model by OpenAI
         GPT_4_0125_PREVIEW (NDLLMProvider): refers to 'gpt-4-0125-preview' model by OpenAI
+        o1_PREVIEW (NDLLMProvider): refers to 'o1-preview' model by OpenAI
+        o1_PREVIEW_2024_09_12 (NDLLMProvider): refers to 'o1-preview-2024-09-12' model by OpenAI
+        o1_MINI (NDLLMProvider): refers to 'o1-mini' model by OpenAI
+        o1_MINI_2024_09_12 (NDLLMProvider): refers to 'o1-mini-2024-09-12' model by OpenAI
```
Comment (acompa) [inline notdiamond/llms/providers.py:29]: As an Enum I'd prefer to see these capitalized by convention:

```suggestion
        O1_PREVIEW (NDLLMProvider): refers to 'o1-preview' model by OpenAI
        O1_PREVIEW_2024_09_12 (NDLLMProvider): refers to 'o1-preview-2024-09-12' model by OpenAI
        O1_MINI (NDLLMProvider): refers to 'o1-mini' model by OpenAI
        O1_MINI_2024_09_12 (NDLLMProvider): refers to 'o1-mini-2024-09-12' model by OpenAI
```

File: notdiamond/llms/providers.py [hunk 2]
```diff
@@ -86,6 +89,10 @@ class NDLLMProviders(Enum):
     GPT_4o_MINI_2024_07_18 = ("openai", "gpt-4o-mini-2024-07-18")
     GPT_4o_MINI = ("openai", "gpt-4o-mini")
     GPT_4_0125_PREVIEW = ("openai", "gpt-4-0125-preview")
+    o1_PREVIEW = ("openai", "o1-preview")
+    o1_PREVIEW_2024_09_12 = ("openai", "o1-preview-2024-09-12")
+    o1_MINI = ("openai", "o1-mini")
+    o1_MINI_2024_09_12 = ("openai", "o1-mini-2024-09-12")
```
Comment (acompa) [inline notdiamond/llms/providers.py:95]: Same as above:

```suggestion
    O1_PREVIEW = ("openai", "o1-preview")
    O1_PREVIEW_2024_09_12 = ("openai", "o1-preview-2024-09-12")
    O1_MINI = ("openai", "o1-mini")
    O1_MINI_2024_09_12 = ("openai", "o1-mini-2024-09-12")
```

File: notdiamond/settings.py [hunk 3]
```diff
@@ -19,7 +19,7 @@
 
 
 NOTDIAMOND_API_URL = os.getenv(
-    "NOTDIAMOND_API_URL", "https://not-diamond-server.onrender.com"
+    "NOTDIAMOND_API_URL", "https://api.notdiamond.ai"
```
Comment (acompa) [inline notdiamond/settings.py:22]: Did I mention making this change? I forgot whether I had told you!

File: poetry.lock [hunk 4]
```diff
@@ -4051,10 +4051,54 @@ description = "Database Abstraction Library"
 optional = true
 python-versions = ">=3.7"
 files = [
+    {file = "SQLAlchemy-2.0.34-cp310-cp310-macosx_10_9_x86_64.whl", hash = "sha256:95d0b2cf8791ab5fb9e3aa3d9a79a0d5d51f55b6357eecf532a120ba3b5524db"},
```
Comment (acompa) [inline poetry.lock:4054]: Did you intend to include these wheels here?

**Review by acompa — COMMENTED** [review]
Clarifying some stylistic q's, but otherwise only want to block until we confirm including the wheels for `SQLAlchemy`.

File: poetry.lock (hunk 4 above)
Comment (r0ymanesco) [reply poetry.lock:4054]: This must have been added when I updated openai. I don't know why it was ever a dependency but clearly it was already there before I ran the update

Comment (r0ymanesco) [reply poetry.lock:4054]: It's actually a dependency of langchain

File: notdiamond/settings.py (hunk 3 above)
Comment (r0ymanesco) [reply notdiamond/settings.py:22]: Saw it on readme

File: notdiamond/llms/providers.py (hunk 1 above)
Comment (r0ymanesco) [reply notdiamond/llms/providers.py]: Done

File: notdiamond/llms/providers.py (hunk 2 above)
Comment (r0ymanesco) [reply notdiamond/llms/providers.py]: Done

**Review by acompa — APPROVED** [review]

**Pull request merged**